controllers/clustermodule_reconciler.go

```diff
@@ -162,6 +162,11 @@ func (r Reconciler) toAffinityInput(obj client.Object) []reconcile.Request {
 		return nil
 	}
 
+	if cluster.Spec.InfrastructureRef == nil {
```

Comment:
A bunch of these tests are failing for me locally with or without these changes - not sure what the root cause is.

Reply:
Pulled them locally, they have passed. Yassine triggered a retry, should work out.